{"name": "test_product_image", "status": "failed", "statusDetails": {"message": "AssertionError: Cant find element by locator: ('css selector', '#content > div > div.col-sm-8 > ul.thumbnails > li:nth-child(3) > a > img')", "trace": "self = <page_objects.ProductPage.ProductPage object at 0x105b2d700>, locator = ('css selector', '#content > div > div.col-sm-8 > ul.thumbnails > li:nth-child(3) > a > img')\n\n    def _verify_element_presence(self, locator: tuple):\n        try:\n>           return WebDriverWait(self.browser, self.browser.t).until(EC.visibility_of_element_located(locator))\n\npage_objects/BasePage.py:22: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <selenium.webdriver.support.wait.WebDriverWait (session=\"103ba9c1f6db40eb5bef56bbda018b27\")>\nmethod = <function visibility_of_element_located.<locals>._predicate at 0x105b581f0>, message = ''\n\n    def until(self, method, message=''):\n        \"\"\"Calls the method provided with the driver as an argument until the \\\n        return value does not evaluate to ``False``.\n    \n        :param method: callable(WebDriver)\n        :param message: optional message for :exc:`TimeoutException`\n        :returns: the result of the last call to `method`\n        :raises: :exc:`selenium.common.exceptions.TimeoutException` if timeout occurs\n        \"\"\"\n        screen = None\n        stacktrace = None\n    \n        end_time = time.time() + self._timeout\n        while True:\n            try:\n                value = method(self._driver)\n                if value:\n                    return value\n            except InvalidSelectorException as e:\n                raise e\n            except self._ignored_exceptions as exc:\n                screen = getattr(exc, 'screen', None)\n                stacktrace = getattr(exc, 'stacktrace', None)\n            time.sleep(self._poll)\n            if time.time() > end_time:\n                break\n>       raise TimeoutException(message, screen, stacktrace)\nE       selenium.common.exceptions.TimeoutException: Message: \nE       Stacktrace:\nE       0   chromedriver                        0x00000001005cad98 chromedriver + 4337048\nE       1   chromedriver                        0x00000001005c2e14 chromedriver + 4304404\nE       2   chromedriver                        0x00000001001efa5c chromedriver + 293468\nE       3   chromedriver                        0x0000000100234d50 chromedriver + 576848\nE       4   chromedriver                        0x000000010026f908 chromedriver + 817416\nE       5   chromedriver                        0x0000000100228a5c chromedriver + 526940\nE       6   chromedriver                        0x0000000100229908 chromedriver + 530696\nE       7   chromedriver                        0x0000000100590de4 chromedriver + 4099556\nE       8   chromedriver                        0x00000001005952a0 chromedriver + 4117152\nE       9   chromedriver                        0x000000010059b52c chromedriver + 4142380\nE       10  chromedriver                        0x0000000100595da0 chromedriver + 4119968\nE       11  chromedriver                        0x000000010056da74 chromedriver + 3955316\nE       12  chromedriver                        0x00000001005b2a48 chromedriver + 4237896\nE       13  chromedriver                        0x00000001005b2bc4 chromedriver + 4238276\nE       14  chromedriver                        0x00000001005c2a8c chromedriver + 4303500\nE       15  libsystem_pthread.dylib             0x000000018cafffa8 _pthread_start + 148\nE       16  libsystem_pthread.dylib             0x000000018cafada0 thread_start + 8\n\nvenv/lib/python3.9/site-packages/selenium/webdriver/support/wait.py:89: TimeoutException\n\nDuring handling of the above exception, another exception occurred:\n\nbrowser = <selenium.webdriver.remote.webdriver.WebDriver (session=\"103ba9c1f6db40eb5bef56bbda018b27\")>\n\n    def test_product_image(browser):\n        MainPage(browser).go_to_product_page()\n>       image_tap = ProductPage(browser).click_image_tap()\n\ntests/test_opencart.py:96: \n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \npage_objects/ProductPage.py:38: in click_image_tap\n    image_tap = self._element(self.IMAGE_TAP)\npage_objects/BasePage.py:27: in _element\n    return self._verify_element_presence(locator)\n_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ \n\nself = <page_objects.ProductPage.ProductPage object at 0x105b2d700>, locator = ('css selector', '#content > div > div.col-sm-8 > ul.thumbnails > li:nth-child(3) > a > img')\n\n    def _verify_element_presence(self, locator: tuple):\n        try:\n            return WebDriverWait(self.browser, self.browser.t).until(EC.visibility_of_element_located(locator))\n        except TimeoutException:\n>           raise AssertionError(\"Cant find element by locator: {}\".format(locator))\nE           AssertionError: Cant find element by locator: ('css selector', '#content > div > div.col-sm-8 > ul.thumbnails > li:nth-child(3) > a > img')\n\npage_objects/BasePage.py:24: AssertionError"}, "start": 1695470400852, "stop": 1695470406317, "uuid": "e55b269f-d239-47cf-a127-e2b18d621f41", "historyId": "00e447fed6f0b3fcc93d0081decf7990", "testCaseId": "00e447fed6f0b3fcc93d0081decf7990", "fullName": "tests.test_opencart#test_product_image", "labels": [{"name": "parentSuite", "value": "tests"}, {"name": "suite", "value": "test_opencart"}, {"name": "host", "value": "Karinas-MacBook-Air.local"}, {"name": "thread", "value": "61930-MainThread"}, {"name": "framework", "value": "pytest"}, {"name": "language", "value": "cpython3"}, {"name": "package", "value": "tests.test_opencart"}]}